src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,110 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
+import { getPostDetails } from '../../api/posts';
+import
+{ getPostComments, sendComment, deleteComment } from '../../api/comments';
+
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ selectedPostId }) => {
+  const [post, setPost] = useState({});
+  const [comments, setComments] = useState([]);
+  const [commentsVisibility, setCommentsVisibility] = useState(true);
+
+  const commentsHandle = () => {
+    setCommentsVisibility(!commentsVisibility);
+  };
+
+  useEffect(() => {
+    const getDetails = async() => {
+      const selectedPost = await getPostDetails(selectedPostId);
+      const postComments = await getPostComments(selectedPostId);
+
+      setPost(selectedPost.data);
+      setComments(postComments.data);
+    };
+
+    getDetails();
+  }, [selectedPostId]);
+
+  const getDetails = async() => {
+    const selectedPost = await getPostDetails(selectedPostId);
+    const postComments = await getPostComments(selectedPostId);
+
+    setPost(selectedPost.data);
+    setComments(postComments.data);
+  };
```

Comment:
No need to duplicate here this function. You use it later when adding or deleting comments, but all you need - is to re-fetch new comments after deleting or adding a new one.
Read comments below